### PR TITLE
#162725268 Add article pagination support

### PR DESCRIPTION
**Pull request by mariamiah**
### What does this PR do?
This PR adds pagination support for the articles

### Description of Task to be completed?
Numerous articles make it hard for the reader to scroll down the screen. This PR is meant to improve article readability by introducing pages to the articles.

### How should this be manually tested?
- fetch this branch by running the command `git fetch origin ft-article-pagination-support-162725268` in the terminal.
- check into the branch `git checkout ft-article-pagination-support-162725268`
- start the virtual environment `source venv/bin/activate`
- re-install requirements `pip install -r requirements.txt`
- start the application `python manage.py runserver`
- Run postman and test the links as shown in the screenshots below
- Set the number of articles for each page using `api/articles/?limit=<integer>&page=<integer>`

### What are the relevant pivotal tracker stories?
[#162725268](https://www.pivotaltracker.com/n/projects/2232225/stories/162725268)

### Screen shots
screen shot 2019-01-22 at 4 10 54 pm

### Postman
[Run in Postman](https://app.getpostman.com/run-collection/4000a40852668974b80d)

### CODE REVIEW

**Comment by mariamiah** [comment]
> Are the next and previous fields added automatically in response body
Yes, adding the pagination class to the view, implements these fields automatically @MuhweziDeo

**Review by armstrongsouljah — COMMENTED** [review]
Looking the pagination class, do you have control over how many articles appear on a single page?

**Comment by mariamiah** [comment]
@armstrongsouljah  i have implemented LimitOffsetPagination, in order to allow for control over articles per page

**Review by malfahad — DISMISSED** [review]
Looks and works great.

**Comment by ja-odur** [comment]
@mariamiah How do I set the number of articles per page.

**Comment by ja-odur** [comment]
@mariamiah your returned url `http://localhost:8000/api/articles/?limit=1&offset=1` has `offset` instead of page. Please change this to use `page` instead.

File: authors/apps/articles/paginators.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from rest_framework.pagination import LimitOffsetPagination
+
+
+class ArticleLimitOffSetPagination(LimitOffsetPagination):
+    default_limit = 2
+    max_limit = 10
```
Comment (ja-odur) [inline authors/apps/articles/paginators.py]: @mariamiah Please do not set the max limit and default limit could be set to 10

Comment (mariamiah) [reply authors/apps/articles/paginators.py]: @ja-odur I have fixed it

**Review by d-rita — DISMISSED** [review]

**Comment by ja-odur** [comment]
@mariamiah the page numbering seems to be starting from zero. can you adjust this to start from 1

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -2,21 +2,24 @@
...
+    check_if_is_author,
+)
+from .paginators import ArticleLimitOffSetPagination
 
 
 class ArticleListCreateView(generics.ListCreateAPIView):
     queryset = Article.objects.all()
     serializer_class = ArticleSerializer
-    permission_classes = (IsAuthenticated,)
+    # permission_classes = (IsAuthenticated,)
```
Comment (d-rita) [inline authors/apps/articles/views.py]: Commented code should be removed

**Review by d-rita — DISMISSED** [review]
LGTM

**Review by ja-odur — APPROVED** [review]
LGTM

**Review by calvinpete — COMMENTED** [review]
Could you please use markdown syntax to indicate that **source venv/bin/activate** and **pip install -r requirements.txt** are command lines.

**Review by d-rita — APPROVED** [review]

**Review by MuhweziDeo — APPROVED** [review]

**Review by kdaglas — APPROVED** [review]

**Pull request merged**